core/pom.xml

```diff
@@ -83,6 +83,11 @@
       <groupId>org.hdrhistogram</groupId>
       <artifactId>HdrHistogram</artifactId>
     </dependency>
+    <dependency>
+      <groupId>net.jcip</groupId>
+      <artifactId>jcip-annotations</artifactId>
+      <optional>true</optional>
+    </dependency>
```

Comment:
I went with this artifact for simplicity, it's the only one that has everything and doesn't pull a bunch of other stuff.

Optional so that it's not pulled in client applications. As far as I can tell, the only consequence is that the annotations appear in error if you browse the sources from the dependent project (this happens with Guava too).